apps/vs-code-designer/src/assets/UnitTestTemplates/TestClassFile

```diff
@@ -1,77 +1,128 @@
         {
-            // Arrange – define mocks either from a static source or as a lambda function
-            var triggerMock = new OperationMock("When_a_HTTP_request_is_received", FlowStatus.Succeeded, new JObject()
-            {
-                { "body", new JObject() { { "isReleased", true }, { "isEnabled", false }, { "isFlag", false } } }
-            });
-
-            // Act 
-            var unitTestFlowRun = this.UnitTestExecutor.ExecuteWorkflow(this.WorkflowDefinition, this.Mocks, triggerMock).Result;
-
-            // Assert
-            Assert.IsNotNull(unitTestFlowRun);
-            Assert.AreEqual(FlowStatus.Succeeded, unitTestFlowRun.Properties.Status);
-            Assert.AreEqual("myvalue", unitTestFlowRun.Properties.Actions["Get_Rows_Receipt"].Outputs.Value<string>());
+            //Get the root path of the workspace
+            this.rootPath = Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, @"..\..\..\..\.."));
+
+            // Build the full paths to the required JSON files.
+            this.workflowDefinitionPath = Path.Combine(rootPath, "<%= LogicAppName %>", "<%= WorkflowName %>", "workflow.json");
+            this.connectionsPath = Path.Combine(rootPath, "<%= LogicAppName %>", "connections.json");
+            this.parametersPath = Path.Combine(rootPath, "<%= LogicAppName %>", "parameters.json");
+            this.localSettingsPath = Path.Combine(rootPath, "<%= LogicAppName %>", "local.settings.json");
+
+            // Set path to the mock.json file.
+            var mockDataPath = Path.Combine(rootPath, "Tests", "<%= LogicAppName %>", "<%= WorkflowName %>", "<%= UnitTestName %>", "<%= UnitTestName %>-mock.json");
+
+            // Load the mock data from the corrected path.
+            var mockData = JObject.Parse(File.ReadAllText(mockDataPath));
+            this.triggerMock = mockData["triggerMocks"].ToObject<Dictionary<string, TriggerMock>>().FirstOrDefault().Value;
+            this.actionMocks = mockData["actionMocks"].ToObject<Dictionary<string, ActionMock>>();
         }
 
+        /// <summary>
+        /// A sample unit test for executing the workflow named <%= WorkflowName %>.
+        /// This method shows how to set up mock data, execute the workflow, and assert the outcome.
+        /// </summary>
         [TestMethod]
-        public void ExecuteWorkflow_WithMockAction_ShouldSucceed_Dynamic()
+        public async Task <%= WorkflowName %>_<%= UnitTestName %>_ExecuteWorkflow_SUCCESS()
         {
-            // Arrange – define mocks either from a static source or as a lambda function
-            var triggerMock = this.Mocks["When_a_HTTP_request_is_received"];
-            this.Mocks["Get_Rows_Receipt"] = new OperationMock("Get_Rows_Receipt", FlowStatus.Succeeded, this.MockDynamicActionOutputCallback);
+            // PREPARE Mock
+            // var actionMocks = this.actionMocks;
+            // actionMocks["<actionName>"] = new ActionMock(
+            //     "<actionName>",
```

Comment:
can we add named parameters here for better readablity

Reply:
fixed